Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -1193,9 +1193,11 @@ PredefinedColorSpace WebGLRenderingContextBase::drawingBufferColorSpace() const
 
 void WebGLRenderingContextBase::setDrawingBufferColorSpace(PredefinedColorSpace colorSpace)
 {
-    if (m_drawingBufferColorSpace != colorSpace) {
+    if (isContextLost())
+        return;
+
+    if (m_drawingBufferColorSpace != colorSpace && m_context->setDrawingBufferColorSpaceIfSupported(toDestinationColorSpace(colorSpace))) {
         m_drawingBufferColorSpace = colorSpace;
```

Comment:
I'm not a fan of having two pieces of state that need to be kept in sync. Is it possible to have GraphicsContextGL be the source of truth and remove the state mirroring from `WebGLRenderingContextBase`?